src/views/dialogs/ReferralDialog.tsx

```diff
@@ -36,15 +36,20 @@ const CONTENT_SECTIONS = [
 export const ReferralDialog = ({ setIsOpen, refCode }: DialogProps<ReferralDialogProps>) => {
   const stringGetter = useStringGetter();
   const { dydxAddress } = useAccounts();
-  const { data: referralAddress, isFetched: isReferralAddressFetched } =
-    useReferralAddress(refCode);
+  const {
+    data: referralAddress,
+    isPending: isReferralAddressPending,
+    isFetched: isReferralAddressFetched,
+  } = useReferralAddress(refCode);
   const {
     affiliateMetadataQuery: { data: affiliatesInfo, isSuccess: isAffiliatesInfoSuccess },
   } = useAffiliatesInfo(referralAddress);
 
   const isNotEligible = isAffiliatesInfoSuccess && !affiliatesInfo.isEligible;
   const invalidReferralCode = isReferralAddressFetched && !referralAddress;
 
+  if (isReferralAddressPending) return null;
```

Comment:
doing this to prevent screen flickering when the referral info actually loads

Reply:
is this better than using `<LoadingSpace>`?

Reply:
yeah because this is just a dialog so ideally the user can view the rest of the site uninterrupted while the info for dialog loads